# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1482949993 18000
#      Wed Dec 28 13:33:13 2016 -0500
# Node ID 8bc5d69a96f31d84f3f01e3f3941740933acccdb
# Parent  d244af1d0bc7a21b8927f4cea8e21cd596ef73ac
* fCMatrix.cc: Fix omission from previous change.

diff --git a/liboctave/array/fCMatrix.cc b/liboctave/array/fCMatrix.cc
--- a/liboctave/array/fCMatrix.cc
+++ b/liboctave/array/fCMatrix.cc
@@ -2524,17 +2524,17 @@ FloatComplexMatrix::lssolve (const Float
       Array<F77_INT> iwork (dim_vector (liwork, 1));
       F77_INT* piwork = iwork.fortran_vec ();
 
       F77_INT tmp_info = 0;
       F77_INT tmp_rank = 0;
 
       F77_XFCN (cgelsd, CGELSD, (m, n, nrhs, F77_CMPLX_ARG (tmp_data), m,
                                  F77_CMPLX_ARG (pretval), maxmn,
-                                 ps, rcon, rank, F77_CMPLX_ARG (work.fortran_vec ()),
+                                 ps, rcon, tmp_rank, F77_CMPLX_ARG (work.fortran_vec ()),
                                  lwork, prwork, piwork, tmp_info));
 
       info = tmp_info;
       rank = tmp_rank;
 
       // The workspace query is broken in at least LAPACK 3.0.0
       // through 3.1.1 when n >= mnthr.  The obtuse formula below
       // should provide sufficient workspace for ZGELSD to operate
